### PR TITLE
Cross Sell: Cards should be links

### LINKED ISSUE #924: Frequently Bought together: products should be links
I don't know if this was discussed when the feature was made, but I think at least *some* portion of the product card ought to be a link that takes you to a product page, and it should be obvious which part.

image

I really doubt that people will add something to their cart without even being able to go to the product page

CC @sterlinghirsh @federicobadini

### PR DESCRIPTION
**Pull request by masonmcelvain**
Makes the cross sell product cards links. Clicking the checkbox will select/deselect the cross sell product.

Note that `onKeyUp` was changed to `onKeyDown`. Without this change, clicking "Enter" with the checkbox focused caused the link to be followed (and the checkbox to be toggled). The expected behavior is to toggle the checkbox and not follow the link, and using `onKeyDown` achieves that.

Next link docs: https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag

## CR

Easiest to view with whitespace hidden.

## QA

Make sure the cards are now links, and that you are still able to toggle the selections as before using the checkbox.

Closes #924

### CODE REVIEW

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Attempting to add out of stock items to the cart results in a console error:
image

https://user-images.githubusercontent.com/5289715/204681516-473934ea-002f-49a4-a5a9-779a84f70a05.mp4


I was still able to add the cross sold item on master, even though the main item was out of stock.

**Comment by masonmcelvain** [comment]
I'm seeing the same behavior on main (react-commerce.vercel.app on the right) for `/products/xbox-series-x-power-and-eject-button-board` as this branch (on the left).

image

[Screencast from 11-30-2022 09:48:46 AM.webm](https://user-images.githubusercontent.com/52104630/204827191-07cb9d16-c977-4c63-a874-1b1aed659a66.webm)

un_dev_block ✌🏻 Lmk if you're still seeing the same behavior, it's not obvious from your console screenshots what went wrong there 🤔

**Comment by deltuh-vee** [comment]
QA 🎬 
Cross sell cards are now links, and you are still able to toggle the selections as before using the checkbox.

**Review by ardelato — APPROVED** [review]
CR 👍 but deploy_block 🎨 on minor UI suggestion

We should add a hover effect around the checkboxes to inform the user that a different action will be performed when clicking the checkbox vs clicking the card.


Here I just set the `opacity` to `50%` when hovering over the checkbox.

https://user-images.githubusercontent.com/22064420/205997446-9d0573f2-ea67-421a-9996-b3975934c89a.mp4

File: frontend/templates/product/sections/CrossSellSection/index.tsx [hunk 1]
```diff
@@ -89,11 +91,17 @@ export function CrossSellSection({
...
    const currentProduct = React.useMemo<CrossSellProduct>(() => {
       return {
+         handle: product.handle,
          title: product.title,
          rating: product?.rating?.value,
          reviewsCount: product?.reviewsCount,
       };
-   }, [product?.rating?.value, product?.reviewsCount, product.title]);
+   }, [
+      product?.handle,
```
Comment (danielcliu-ifixit) [inline frontend/templates/product/sections/CrossSellSection/index.tsx:100]: do we need the `?`

**Comment by erinemay** [comment]
>We should add a hover effect around the checkboxes to inform the user that a different action will be performed when clicking the checkbox vs clicking the card.
>
>Here I just set the opacity to 50% when hovering over the checkbox.

I agree that the usability here is a little odd, but I think the universal this-is-a-clickable-action cue is that a button becomes darker. I think making it more opaque usually sends the message that that item is non-clickable.

**Comment by deltuh-vee** [comment]
dev_block ✊ 
It might be better to have the product title be an underlined link and have the rest of the card toggle the selection for a product. Thoughts?

**Comment by danielcliu-ifixit** [comment]
CR :v:

**Review by danielcliu-ifixit — APPROVED** [review]

**Comment by erinemay** [comment]
> dev_block ✊ It might be better to have the product title be an underlined link on hover and have the rest of the card toggle the selection for a product so we don't deviate too much from the current behavior. Thoughts?

@deltuh-vee, I agree, but... From the issue: 
> Looking at Amazon, the entire card links to the product, and the only the checkbox let's you deselect the product. We might do the same. Thoughts?

>E-commerce wants to match Amazon closely.

**Comment by erinemay** [comment]
>1.) awkwardly opting them into second purchase by selecting it be default 2.) having it default to selecting the current product as well by default. This is fine in concept but i could easily see someone adding to cart at the top, then scrolling down and seeing a useful addition in the Cross Sell section and hitting the button there too, without realizing they now have 2 of the current product in their cart.

I agree here too. I think I remember a conversation about this. It worked ok on the old product pages because cross sells were way higher up. When you got to the page, you could make your cross-sell decision without scrolling. 

From what I remember, we put a pin in either bumping up the cross-sell on the page or mucking with the behavior. @sterlinghirsh, do you remember the outcome of that decision?

File: frontend/templates/product/sections/CrossSellSection/index.tsx [hunk 2]
```diff
@@ -303,137 +313,155 @@ function CrossSellItem({
...
+                           onClick={(event) => {
+                              onChange(!isSelected);
+                              event.preventDefault();
+                              event.stopPropagation();
                            }}
-                           top={{
-                              base: 'auto',
-                              md: 4,
+                           onKeyDown={(event) => {
+                              if (['Enter'].includes(event.code)) {
```
Comment (danielbeardsley) [inline frontend/templates/product/sections/CrossSellSection/index.tsx:427]: Is there a reason for this odd condition? It seems an odd way to do an equality test.

Comment (masonmcelvain) [reply frontend/templates/product/sections/CrossSellSection/index.tsx:427]: Good eye I didn't notice that. Looking at other examples of this pattern in this repo, it seems to be used when we expect to add another value to the array in the future. Given that we probably wouldn't add any other keypress to this array, I agree that checking equality with `===` makes more sense.

**Comment by masonmcelvain** [comment]
Wow lots of activity! Thanks for all the feedback.

I ended up following the default checkbox hover pattern that Chakra uses.

un_dev_block ✌🏻 un_deploy_block ✌🏻

**Comment by deltuh-vee** [comment]
QA 🎬 
* Cross sell Cards are now links
* You are still able to toggle the selections as before using the checkboxes
* Checkboxes will dim a little bit when hovered over

**Pull request merged**